flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -410,15 +410,54 @@ SqlShowViews SqlShowViews() :
 }
 
 /**
-* Parse a "Show Tables" metadata query command.
+* SHOW TABLES FROM [catalog.] database sql call.
 */
 SqlShowTables SqlShowTables() :
 {
+    SqlIdentifier databaseName;
+    SqlCharStringLiteral likeLiteral = null;
+    String prep = "FROM";
+    boolean notLike = false;
+    SqlParserPos pos;
 }
 {
     <SHOW> <TABLES>
+    [
+        [( <FROM> | <IN> { prep = "IN"; } )]
+        { pos = getPos(); }
+        databaseName = CompoundIdentifier()
+        [
+            [
+                <NOT>
+                {
+                    notLike = true;
+                }
+            ]
+            <LIKE>  <QUOTED_STRING>
+            {
+                String likeCondition = SqlParserUtil.parseString(token.image);
+                likeLiteral = SqlLiteral.createCharString(likeCondition, getPos());
+            }
+        ]
+        {
+            return new SqlShowTables(pos, prep, databaseName, notLike, likeLiteral);
+        }
```

Comment:
It seems this part is redundant with the following NOT LIKE parser, could you combine them?

Reply:
we could make it simpler, done :)